File: src/main/java/br/com/warnit/repository/LocalidadeRepository.java
package br.com.warnit.repository;

import br.com.warnit.model.domain.Localidade;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

/**
 * @author Guilherme Camargo
 * @since 01/02/2019
 * @version 1.0
 * */
@Repository
public interface LocalidadeRepository extends JpaRepository<Localidade, Long> {

    @Transactional(readOnly = true)
    Optional<Localidade> findByCep(String cep);
}

File: WarnIt-Address/src/main/kotlin/br/com/warnit/address/repository/LocalidadeRepository.kt
package br.com.warnit.address.repository

import br.com.warnit.address.model.Localidade
import org.springframework.data.repository.CrudRepository
import org.springframework.transaction.annotation.Transactional
import java.util.*


interface LocalidadeRepository: CrudRepository<Localidade, Long>{
    @Transactional(readOnly = true)
    fun findByCep(cep: String): Optional<Localidade>
}